contracts/src/v0.8/dev/vrf/VRFV2PlusWrapper.sol

```diff
@@ -384,11 +388,27 @@ contract VRFV2PlusWrapper is ConfirmedOwner, TypeAndVersionInterface, VRFConsume
     lastRequestId = requestId;
   }
 
+  function checkPaymentMode(bytes memory extraArgs, bool isLinkMode) public pure {
```

Comment:
note that extraArgs can be empty and we should treat it as a valid extraArg and accept the request as link payment. Can we add a test case for this scenario? we may need to handle `extraArgs.length == 0` case separately. https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/dev/vrf/VRFCoordinatorV2_5.sol#L200

Reply:
Ah I forgot about this, is this done for backwards compatibility reasons?

Reply:
Wouldn't it be easier to always enforce that extraArgs is always provided?

Reply:
Yeah it might simplify our code slightly and allow marginal gas savings. Rationale for allowing empty bytes was that extraArgs feels like an optional field that customers may or may not provide. But if we assume that all consumers will be using our consumer contracts, requiring non-zero `extraArgs` doesn't add much complexity to customers.

Reply:
Fixed this by adding check for empty extraArgs